zipline/pipeline/term.py

```diff
@@ -307,7 +220,128 @@ def _compute(self, windows, dates, assets, mask):
         return out
 
 
-class AssetExists(Term):
+class AtomicTerm(Term):
```

Comment:
There are a couple places where we do `if not term.atomic: <iterate over term.inputs>`.  I wonder if those would be made cleaner if `AtomicTerm` just defined `inputs` as an empty tuple?